packages/reports/addon/components/common-actions/schedule.ts

```diff
@@ -224,14 +251,29 @@ export default class ScheduleActionComponent extends Component<Args> {
    */
   @action
   async onOpen() {
-    //Kick off a fetch for existing delivery rules
-    this.deliveryRule = this.args.model.deliveryRuleForUser;
-
     try {
-      const rule = await this.deliveryRule;
-      this.localDeliveryRule = rule ? rule : this.localDeliveryRule || this._createNewDeliveryRule();
+      this.deliveryRules = await this.args.model.deliveryRulesForUser;
+      if (this.deliveryRules) {
+        this.localDeliveryRule = this.deliveryRules[0] ? this.deliveryRules[0] : this.localDeliveryRule || undefined;
+        if (this.localDeliveryRule) {
+          if (
+            this.localDeliveryRule.format.type !== 'pdf' &&
+            this.localDeliveryRule.delivery !== 'none' &&
+            this.localDeliveryRule.name
+              .toString()
+              .match(/\w+ \bdelivered\b +\w+ \bevery\b (day|week|month|quarter|year)$/)
+          ) {
+            this.localDeliveryRule.name = `Email delivered ${this.localDeliveryRule.format.type} every week`;
```

Comment:
I think it might be easier to not store the generated string. Can we generate the string on the client in the case a user defined name is empty?

Reply:
So whenever the customer sets the delivery rule name to an empty string we just pop in the default? And by stored on client what do you mean?

Reply:
Yes, when the name `isBlank` render the generated name in the component.